modules/gateway.go

```diff
@@ -37,19 +82,22 @@ type Gateway interface {
 	// known peers.
 	RelayTransaction(consensus.Transaction) error
 
-	// AddMe is the RPC version of AddPeer. It is assumed that the supplied
-	// peer is the peer making the RPC.
-	AddMe(network.Address) error
-
-	// SendBlocks is an RPC that returns a set of sequential blocks following
-	// the most recent known block ID in of the 32 IDs provided. The number of
-	// blocks returned is unspecified.
-	SendBlocks([32]consensus.BlockID) ([]consensus.Block, error)
-
-	// SharePeers is an RPC that returns a set of the Gateway's peers. The
-	// number of peers returned is unspecified.
-	SharePeers() ([]network.Address, error)
-
 	// Info reports metadata about the Gateway.
 	Info() GatewayInfo
 }
+
+// ReaderRPC returns a closure that can be passed to Gateway.RPC to read a
+// single value.
+func ReaderRPC(obj interface{}, maxLen uint64) RPCFunc {
+	return func(conn NetConn) error {
+		return conn.ReadObject(obj, maxLen)
+	}
+}
+
+// WriterRPC returns a closure that can be passed to Gateway.RPC to write a
+// single value.
+func WriterRPC(obj interface{}) RPCFunc {
+	return func(conn NetConn) error {
+		return conn.WriteObject(obj)
+	}
+}
```

Comment:
Are these two functions used extensively? Because this file is full of stuff that we're going to want to support through the entirety of every major version.

I think this is more okay because it's entirely dependent on other code in this file. It'll work as long all the other parts of this file stay the same, and the whole worry is related to maintaining things outside of this file. It shouldn't be any more work to maintain the project whether this code is here or not.